src/components/modals/CategoryFilterModal.vue

```diff
@@ -0,0 +1,155 @@
+                <div class="field has-addons" v-if="selectedCategories.length > 0">
+                    <div class="control" v-for="(key, index) in selectedCategories" :key="`${key}-${index}`">
+                        <span class="block margin-right">
+                            <a href="#" @click="unselectCategory(key)">
+                                <span class="tags has-addons">
+                                    <span class="tag">{{ key }}</span>
+                                    <span class="tag is-danger">
+                                        <i class="fas fa-times"></i>
+                                    </span>
+                                </span>
+                            </a>
+                        </span>
+                    </div>
+                </div>
+                <div class="field has-addons" v-else>
+                    <span class="tags">
+                        <span class="tag">No categories selected</span>
+                    </span>
+                </div>
+            </div>
+            <hr/>
+            <div>
+                <input
+                    v-model="allowNsfw"
+                    id="nsfwCheckbox"
+                    class="is-checkradio has-background-color"
+                    type="checkbox"
+                    :class="[{'is-dark': !isDarkTheme, 'is-white': isDarkTheme}]"
+                >
+                <label for="nsfwCheckbox">Allow NSFW (potentially explicit) mods</label>
+            </div>
+            <br/>
+            <div>
+                <div v-for="(key, index) in categoryFilterValues" :key="`cat-filter-${key}-${index}`">
+                    <input
+                        name="categoryFilterCondition"
+                        type="radio"
+                        :id="`cat-filter-${key}-${index}`"
+                        :value=key
+                        :checked="index === 0 ? true : undefined" v-model="categoryFilterMode"
+                    />
+                    <label :for="`cat-filter-${key}-${index}`">
+                        <span class="margin-right margin-right--half-width" />
+                        {{ key }}
+                    </label>
+                </div>
+            </div>
+        </template>
+        <template v-slot:footer>
+            <button class="button is-info" @click="close">
+                Apply filters
+            </button>
+        </template>
+    </modal>
+</template>
+
+<script lang="ts">
+import { Component, Prop, Vue } from "vue-property-decorator";
+
+import { Modal } from '../../components/all';
+import CategoryFilterMode from "../../model/enums/CategoryFilterMode";
+import GameManager from "../../model/game/GameManager";
+import ManagerSettings from "../../r2mm/manager/ManagerSettings";
+
+@Component({
+    components: { Modal }
+})
+export default class CategoryFilterModal extends Vue {
+    activeGame = GameManager.activeGame;
+    settings: ManagerSettings = new ManagerSettings();
```

Comment:
Does this work as intended, as in, do these get cleared from the cached values when appropriate (e.g. game change)?

Reply:
This approach is copied from the Manager page. AFAIK this is just a dummy value so we don't have to check if `settings` is `undefined` in every place where it's called (or use `!` here). The actual value will be overwritten in `created`.

And yes, based on manual testing it works as intended. The whole component will get unmounted/remounted if user wants to change the game. At least for now when the game can only be changed in the `GameSelectionScreen`.